opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -243,8 +246,17 @@ def __hash__(self):
 
 
 _EMPTY_RESOURCE = Resource({})
+_runtime_version = ".".join(map(
+    str,
+    implementation.version[:3]
+    if implementation.version.releaselevel == "final" and not implementation.version.serial
+    else implementation.version
+))
 _DEFAULT_RESOURCE = Resource(
     {
+        PROCESS_RUNTIME_NAME: implementation.name,
+        PROCESS_RUNTIME_VERSION: _runtime_version,
+        PROCESS_RUNTIME_DESCRIPTION: sys.version,
```

Comment:
Please add a test to validate these changes

Reply:
@codeboten would I mock the `sys` object? Or check that those are set in all runtimes that are automated tested (not `NONE`) and rely on the implementation?

Reply:
🤔 good question.... i would probably just check that the versions information in the `Resource` object matches the system's implementation.

Reply:
That is unfortunately not resilient against `serial` and `releaselevel` unless the test re-implements the logic again, which... well...

Reply:
yeah that's a good point... i suppose you could patch those method specifically and use `version` and `implementation` but it seems silly to use patching in only some of the cases. mocking `sys` is probably fine for this test.

Reply:
Well, I had forgotten just how hard is using the `mock` package. I do not know a way to mock that affects the initialization of the package, which is where we access `sys`. Any ideas @codeboten?

Reply:
Try patching the `sys` package that is imported in `opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py`:

``` python
from unittest.mock import patch
from unittest import TestCase

class TestSuite(TestCase):

    @patch("opentelemetry.sdk.resources.sys")
    def test_case(self, mock_sys):
        # configure mock_sys and use it
```

Reply:
Unfortunately that does not help. By the time the mocking occurs, the attributes have already been calculated and added to `_DEFAULT_RESOURCE`.

Reply:
The validation would only need to validate that any new resource created would have process information included correct? In this case you can just override `_DEFAULT_RESOURCE` itself to be anything that you want, and verify that `Resource.create()` has those fields provided.

Reply:
If this is something that would get the PR merged, let’s go.

in reality, I think that the behavior about attaching the releaselevel only if the sequence is zero, is worthy of testing. But my sanity is dwindling ;-)

Reply:
I ended up patching the `_DEFAULT_RESOURCE` in those tests, but two tests fail and figuring out why is beyond my skills. Help? :-)

Reply:
Please rebase so I can see why the tests are failing :)

Reply:
@lzchen done :-)

Reply:
You have regression tests that are failing because you added extra fields to the `_DEFAULT_RESOURCE` in which the tests assume that they don't exist. An easy way to not have to fix all the tests is to use a separate resource like suggested [here](https://github.com/open-telemetry/opentelemetry-python/pull/2660#discussion_r865495059)

Reply:
Yeah well; that the new attributes are to blame, sure. Is the nature of those failures I cannot quite explain, the `test_immutable` above all else.

Reply:
I explained the nature of those failures above. If you take a look at `test_immutable` logic, it's comparing the created `Resource` with an expected resource, but the expected resource does not have the `process.*` attributes. Since the created resource is created from merging the `DEFAULT_RESOURCE`, and because you added new fields to the `DEFAULT_RESOURCE`, the test expects to the newly created resource to have those fields, but the expected "hardcoded" resource does not.

Reply:
As I mentioned below I think process attributes should be a separate resource detector (added bonus with that there won't be regression tests).

Reply:
Done in #a4d5b151